apps/aechannel/test/aesc_fsm_SUITE.erl

```diff
@@ -5181,3 +5252,570 @@ auth_contract_props(btc_auth) ->
         end,
     {"bitcoin_auth", "authorize", Params}.
 
+no_missing_tx(Cfg0) ->
+    Debug = get_debug(Cfg0),
+    Cfg = [?SLOGAN 
+          , {responder_opts, #{keep_running => true } } | Cfg0],
+    assert_empty_msgq(Debug),
+    {I0, R0, Spec0} = channel_spec(Cfg),
+    Spec = Spec0#{responder_opts => #{keep_running => true}},
+    #{ i := #{ channel_id := _ChannelId
+             , fsm := FsmI } = I
+     , r := #{ fsm := _FsmR } = R} = create_channel_from_spec(I0, R0, Spec, ?PORT, Debug, Cfg),
+    ok = rpc(dev1, aesc_fsm, leave, [FsmI]),
+    % Verify leave and FSM timeout
+    {ok, #{info := SignedTx}} = await_leave(I, ?TIMEOUT, Debug),
+    {ok, #{info := SignedTx}} = await_leave(R, ?TIMEOUT, Debug),
+    {ok, _} = receive_info(I, fun died_normal/1, Debug),
+    {ok, _} = receive_from_fsm(info, R, fun(#{info := peer_disconnected}) -> ok end,
+                               ?TIMEOUT, Debug),
+    {ok, #{info := {log, _ILog}}} = receive_log(I, Debug),
+    assert_empty_msgq(Debug),
+    ?LOG(Debug, "queue is empty", []),
+    Info = #{i => I, r => R, spec => Spec},
+    #{i := #{fsm_id := _IFsmId1} = I1, r := #{fsm_id := _RFsmId1} = R1} =
+        reestablish_(Info, SignedTx, ?PORT, Debug),
+    assert_empty_msgq(Debug),
+    shutdown_(I1, R1, Cfg),
+    ok.
+
+missing_close_tx(Cfg0) ->
+    Debug = get_debug(Cfg0),
+    Cfg = [?SLOGAN 
+          , {responder_opts, #{keep_running => true } } | Cfg0],
+    assert_empty_msgq(Debug),
+    {I0, R0, Spec0} = channel_spec(Cfg),
+    Spec = Spec0#{responder_opts => #{keep_running => true}},
+    #{ i := #{ channel_id := _ChannelId
+             , fsm := FsmI } = I
+     , r := #{ fsm := FsmR } = R} = create_channel_from_spec(I0, R0, Spec, ?PORT, Debug, Cfg),
+    ok = rpc(dev1, aesc_fsm, leave, [FsmI]),
+    % Verify leave and FSM timeout
+    {ok, #{info := SignedTx}} = await_leave(I, ?TIMEOUT, Debug),
+    {ok, #{info := SignedTx}} = await_leave(R, ?TIMEOUT, Debug),
+    {ok, _} = receive_info(I, fun died_normal/1, Debug),
+    {ok, _} = receive_from_fsm(info, R, fun(#{info := peer_disconnected}) -> ok end,
+                               ?TIMEOUT, Debug),
+    {ok, #{info := {log, _ILog}}} = receive_log(I, Debug),
+    assert_empty_msgq(Debug),
+    ?LOG(Debug, "queue is empty", []),
+    %% post a close solo while the other party is off
```

Comment:
Up until this point, the code seems to be an excellent candidate for refactoring out a common preamble.